packages/commonwealth/client/scripts/utils/trpcClient.ts

```diff
@@ -1,29 +1,42 @@
-import { httpBatchLink } from '@trpc/client';
+import { createTRPCProxyClient, httpBatchLink } from '@trpc/client';
 import { createTRPCReact } from '@trpc/react-query';
 import type { API } from '../../../server/api/internal-router';
 import { userStore } from '../state/ui/user';
 
-export const trpc = createTRPCReact<API>();
-
 export const BASE_API_PATH = '/api/internal/trpc';
 
+function buildHeaders() {
+  const user = userStore.getState();
+  return {
+    authorization: user.jwt || '',
+    isPWA: user.isOnPWA?.toString(),
+    address:
+      user.addressSelectorSelectedAddress ??
+      user.activeAccount?.address ??
+      user.addresses?.at(0)?.address,
+  };
+}
+
+export const trpc = createTRPCReact<API>();
 export const trpcClient = trpc.createClient({
   links: [
     httpBatchLink({
       url: BASE_API_PATH,
-
-      async headers() {
-        const user = userStore.getState();
-        return {
-          authorization: user.jwt || '',
-          isPWA: user.isOnPWA?.toString(),
-          address:
-            user.addressSelectorSelectedAddress ??
-            user.activeAccount?.address ??
-            user.addresses?.at(0)?.address,
-        };
+      headers() {
+        return buildHeaders();
       },
     }),
   ],
   transformer: undefined,
 });
+
+export const trpcVanillaClient = createTRPCProxyClient<API>({
```

Comment:
can you explain rationale here? I am not sure if this is a good approach to have multiple trpc clients? trpcClient is injected ino the react app in `App.tsx` and trpcVanillaClient is not. I assume that it might have some consequences eg in caching etc, however, I am not sure about it. What is the reason to introduce another client? cc Thoughts @mzparacha @burtonator ?

Reply:
This is just a cleaner way to call the tRPC api, compared to the previous logic it replaces

Reply:
it's cleaner. The problem is that it is additional trpc client that lives in its different world. It has its own cache and it is not triggering react component rerenders. So for me there is no point to introduce it as from FE perspective it might create even more confusion.

Reply:
I agree with @masvelio. In practice we prefer and want to use react-query/trpc hooks instead of vanilla clients, but in some very rare cases we have to use vanilla client and for those I suggest we directly call the endpoint (caching can be easily traced) instead of introducing another query client which we'd to have to manage (separate caching).

I suggest we remove this and use an API call directly instead, we can typecase the response with the return type of the endpoint (just so we don't lose the DX benefit of trpc).

Reply:
I think the rationale behind this new query client is that the deprecated call @Rotorsoft was removing (`DISCOURAGED_NONREACTIVE_fetchProfilesByAddress `) was not yet react-query. So instead of using the react-query trpc client and needing to refactor everything on the frontend, he had created a vanilla one just to replace this use case.

I also agree it should be in react-query, but this might bloat the PR too much. Maybe it is better to just keep it as is and create a ticket for frontend to migrate this query over to react query in the future.